feat: react hook form adapter

deleonio opened the pull request:
## Summary
Added a React Hook Form adapter for KoliBri, enabling seamless integration with the popular React Hook Form library.

## Changes
- Implement React Hook Form adapter for KoliBri components

---

<details>
<summary>🇩🇪 Deutsche Zusammenfassung</summary>

## Zusammenfassung
Ein React-Hook-Form-Adapter für KoliBri hinzugefügt, der eine nahtlose Integration mit der beliebten React-Hook-Form-Bibliothek ermöglicht.

## Änderungen
- React-Hook-Form-Adapter für KoliBri-Komponenten implementiert

</details>

donchi-donald committed:
feat(react hook form adapter): add react form adapter v1 from pr

Refs: #7965

donchi-donald committed:
Merge remote-tracking branch 'origin/develop' into 7965-react-hook-form-adapter

donchi-donald committed:
feat(react hook form adapter): moved all files

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): add @public-ui/react-hook-form-adapter in sample dev dependencies

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): add sample for react hook form

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): update index.tsx in adapter source

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): add story showcasing form with React Hook Form and adapter components

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): add default language

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): add @public-ui/react-hook-form-adapter in .knip.json

Refs: #7965

donchi-donald committed:
Merge branch '7965-react-hook-form-adapter' into 7965-react-hook-form-adapter

Signed-off-by: Donald Donchi Fofack <72475021+donchi-donald@users.noreply.github.com>

deleonio committed:
React-Hook-Form Adapter (#8208)

deleonio committed:
Merge branch 'develop' into 7965-react-hook-form-adapter

donchi-donald committed:
feat(react hook form adapter): update pnpm lock

Refs: #7965

laske185 commented on packages/adapters/react-hook-form-adapter/src/index.tsx:
```diff
@@ -0,0 +1,80 @@
+} from '@public-ui/react';
+
+type ControllerProps<T extends FieldValues> = UseControllerProps<T> & {
+	control?: UseControllerProps<T>['control'];
```
`control` zum Pflichtfeld machen.

laske185 commented on packages/adapters/react-hook-form-adapter/src/index.tsx:
```diff
@@ -0,0 +1,80 @@
+function withController<P extends Record<string, unknown>, T extends FieldValues = FieldValues>(Component: React.ComponentType<P>, valueProp = '_value') {
+	return (props: P & ControllerProps<T>) => {
+		const { name, control, rules, defaultValue, ...rest } = props as any;
+		const context = useFormContext<T>();
```
`useFormContext` entfernen und direkt `control` nutzen, wenn es vorher als Pflichtfeld definiert wurde.

donchi-donald replied:
alles klar, ich passe es gleich an

laske185 reviewed (changes requested):
In der package.json vom sample-react fehlt der react-hook-form-adapter als dependency. Deswegen schlägt die Pipeline fehl.

laske185 commented on packages/samples/react/src/components/react-hook-form/basic.tsx:
```diff
@@ -0,0 +1,155 @@
+	});
+
+	const onSubmit: SubmitHandler<FormData> = (data) => {
+		alert(JSON.stringify(data, null, 2));
```
Beim Submit sollen alle Fehler mit Fehlern markiert werden, die nicht ausgefüllt sind. Aktuell passiert nichts.

donchi-donald replied:
ich korrigiere es

donchi-donald committed:
feat(react hook form adapter): remove useFormContext, add react-hook-form-adapter in package.json in sample-react

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): remove useFormContext, add react-hook-form-adapter in package.json in sample-react

Refs: #7965

donchi-donald committed:
feat(react hook form adapter): sample update

Refs: #7965

laske185 commented on packages/adapters/react-hook-form-adapter/package.json:
```diff
@@ -0,0 +1,83 @@
+{
+        "name": "@public-ui/react-hook-form-adapter",
+        "version": "3.0.2-rc.0",
```
Die Version bitte noch auf die aktuelle Release-Version setzen, damit es beim nächsten Versionshochziehen mit aktualisiert wird.

donchi-donald replied:
hi, ich habe es auf 3.0.4 gesetzt

donchi-donald committed:
feat(react hook form adapter): Update the version to the current release (3.0.4).

Refs: #7965

laske185 committed:
Add build and publish step for react-hook-form adapter

Signed-off-by: Peter Laske <37439758+laske185@users.noreply.github.com>

laske185 committed:
Merge branch 'develop' into 7965-react-hook-form-adapter

# Conflicts:
#	pnpm-lock.yaml

Refs: #7965

laske185 committed:
chore: configure linting and cleanups

Refs: #7965

laske185 committed:
refactor: move react hook form sample to scenarios

Refs: #7965

laske185 committed:
refactor: improve typings

Refs: #7965

laske185 committed:
chore(react-hook-form-adapter): upgrade React and React DOM to version 19.1.1

- Updated dependencies in react-hook-form-adapter package.json to use React 19.1.1 and React DOM 19.1.1.
- Changed import path for components from '@public-ui/react' to '@public-ui/react-v19' in index.tsx.
- Removed required prop from KolInputColorController in basic.tsx.
- Updated pnpm-lock.yaml to reflect the new React version in dependencies.

Refs: #7965

laske185 committed:
fix(react-hook-form-apdater): enhance withController function to support correct types

Refs: #7965

laske185 committed:
Merge branch 'develop' into 7965-react-hook-form-adapter

# Conflicts:
#	packages/samples/react/package.json
#	pnpm-lock.yaml

Refs: #7965

laske185 committed:
fix(samples): update import for KolButton from @public-ui/react-v19

Refs: #7965

laske185 committed:
fix(package): order dependencies

Refs: #7965

laske185 committed:
fix(lerna): order packages

Refs: #7965
